src/platform/renesas/DiagnosticDataProviderImpl.cpp

```diff
@@ -93,6 +95,85 @@ CHIP_ERROR DiagnosticDataProviderImpl::GetBootReason(BootReasonType & bootReason
     return err;
 }
 
+CHIP_ERROR DiagnosticDataProviderImpl::GetNetworkInterfaces(NetworkInterface ** netifpp)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    NetworkInterface *head = nullptr;
+
+    for (Inet::InterfaceIterator interfaceIterator; interfaceIterator.HasCurrent(); interfaceIterator.Next())
+    {
+        char interfaceName[Inet::InterfaceId::kMaxIfNameLength];
+        interfaceIterator.GetInterfaceName(interfaceName, sizeof(interfaceName));
+        struct netif *net_interface = netif_find(interfaceName);
+        if (!net_interface)
+        {
+            continue;
+        }
+
+        NetworkInterface *ifp = new NetworkInterface();
+        if (ifp == nullptr)
+        {
+            err = CHIP_ERROR_NO_MEMORY;
+            break;
+        }
+
+        ifp->name = CharSpan::fromCharString(net_interface->name);
```

Comment:
isn't this the same as interfaceName ? 
It may not be a big thing but I'm trying to get rid of the net_interface all together. It bypasses a layer of abstraction, which is not the best software design

Reply:
Yes, I've updated.